### PR TITLE
games-strategy/freeciv: add 3.1.0_beta2, 9999

### PR DESCRIPTION
**Pull request by Kangie**
Switch to meson build; it'll be the default next version anyway and it's better than autotools (controversial opinion, I know).

### CODE REVIEW

**Comment by Kangie** [comment]
The live ebuild doesn't currently work, but the patch doesn't apply either. freeciv/freeciv#48 has the fix pending upstream and I doubt anyone is chomping at the bit to install the up-to-the-minute game.

**Comment by Kangie** [comment]
TODO for posterity: If someone were to look into the current state of [tolua](https://github.com/LuaDist/tolua) it would be possible to use the system version instead of the bundled version when `USE=system-lua` is enabled.

File: games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		$(meson_use web-server freeciv-web)
+	)
+
+	# Select any client backends that we want enabled; dedicated server shouldn't build a UI
+	# for sanity we'll build the modpack bin with the same UIs as the client.
+	# 'fcmp' = freeciv modpack (client) - gtk3, qt, cli, gtk4
+	if ! use dedicated ; then
+		# there's no SDL modpack backend; rather than incidentally pull in GTK3 (as is default)
+		# let's explicitly set the backend to CLI
+		use sdl && myclient+=( sdl2 ) && use modpack && myfcmp+=( cli )
```
Comment (thesamesam) [inline games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild:112]: What about a helper like this:
```suggestion
		freeciv_enable_flag() {
			local flag=$1
			local client_name=${2:-${flag}}
			local fcmp_name=${3:-${client_name}}
			
			if use ${flag} ; then
				myclient+=( ${client_name} )
				use modpack && myfcmp+=( ${fcmp_name} )
			fi
		}

		freeciv_enable_flag sdl sdl2 cli
		freeciv_enable_flag gtk3 gtk3.22 gtk3
		...
```

File: games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+			use modpack && myfcmp+=( qt )
+		fi
+	else
+		if use modpack ; then
+			myfcmp+=( cli )
+		fi
+	fi
+
+	# the client and fpmc arrays are now populated (or not for dedicated); let's add them to+ emesonargs
+emesonargs+=(
```
Comment (thesamesam) [inline games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild:132]:
```suggestion
	emesonargs+=(
```

File: games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+		-Dclients=$(echo ${myclient[*]} | sed 's/ /,/g')
```
Comment (thesamesam) [inline games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild:133]: Can we do:
-Dclients="${myclient[@]/#/,}"
or so?

Comment (Kangie) [reply games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild:133]: not trivially; that results in `,client1 ,client2 ,...` and makes meson very unhappy.

**Pull request closed without merging**